tests/unit/test_run.py

```diff
@@ -199,5 +178,5 @@ def mock_run(args: list[str], **_: Any) -> subprocess.CompletedProcess[str]:
         results={'username': 'user', 'password': 'pass'},
     )
     assert task.success
-    assert params_path is not None
-    assert not os.path.exists(params_path)
+    assert yaml.safe_load(mock_file.writes[0]) == {'foo': 1, 'bar': ['ab', 'cd']}
```

Comment:
Can we tweak this to:
```suggestion
    assert yaml.safe_load('\n'.join(mock_file.writes)) == {'foo': 1, 'bar': ['ab', 'cd']}
```